### ISSUE TITLE
Narrator is announcing incorrect selection state while navigating to the tab items (Controls, Actions, interactions, Accessibility and performance).[Bug]: 

### ISSUE THREAD

**Issue by msftedad** [storybookjs/storybook#23640]
### Describe the bug

Narrator is announcing incorrect selection state while navigating to the tab items (Controls, Actions, interactions, Accessibility and performance).
Severity-2
Application: Storybook

### To Reproduce

**Pre-requisites:**
Turn on Narrator (Settings-->Accessibility-->Narrator (Or) Ctrl + win key + enter).

**Repro steps:**
1. Open URL: “[Overview - Docs ⋅ Storybook (windows.net)](https://vivacommonstorybooktest.blob.core.windows.net/$web/index.html?path=/docs/overview--docs)” and sign in with valid credentials. 
2. Select “Common navigation with data transform” in the left navigation and invoke it. 
3. Now navigate to ‘Actions’ tab and observe the narrator announcement whether the narrator is narrating correct state or not.

### System

```shell
Windows: Win11 Enterprise
OS: Version 22H2 (OS Build 22621.1992) 
Edge Version: Version 114.0.1823.51 (Official build) (64-bit) 
URL: Overview - Docs ⋅ Storybook (windows.net)
```

### Additional context

**Expected Result:**
Narrator should announce the proper selection state for all the tab items  (Controls, Actions, interactions, Accessibility and performance).>Narrator should announce as selected only for selected tab item and should announce the information only once.

**Attachments:**
Kindly check the video attachment added.

https://github.com/storybookjs/storybook/assets/93735775/5153cec8-9d09-4d71-8138-846fa1b89ebb

![Ellipsis menu-narrator announcement](https://github.com/storybookjs/storybook/assets/93735775/0936512d-641d-4f1b-bebe-c7803816668a)

**Comment by joaoVCNN** [comment]
@vanessayuenn  @msftedad can you assign this for me ? that's my first issue

**Comment by vanessayuenn** [comment]
@joaoVCNN all yours!! Be sure to check the [contribution guide](https://storybook.js.org/docs/react/contribute/how-to-contribute) and hop onto the [Storybook Discord](https://discord.gg/storybook) if you need any help getting started. Looking forward to your contribution! ✨

**Comment by joaoVCNN** [comment]
@vanessayuenn thanks!!

**Comment by msftedad** [comment]
Hi team, Any update on this issue?

**Comment by Sidnioulz** [comment]
This can be bundled into #32235 that lists other ongoing issues with the addon panel tabbar.
